flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManagerDriver.java

```diff
@@ -689,4 +690,54 @@ public void onStopContainerError(ContainerId containerId, Throwable throwable) {
                     throwable);
         }
     }
+
+    public String getConatainerCompletedCause(ContainerStatus containerStatus) {
+        String completeContainerMessage = containerStatus.getDiagnostics();
```

Comment:
Nit, if we make this variable `final` and don't assign it `containerStatus.getDiagnostics()`, then we get a compiler check that every switch case assigns a proper value to this variable. Now with mutable variables this is not easily ensured.